ambry-router/src/test/java/com.github.ambry.router/OperationTrackerTest.java

```diff
@@ -159,6 +161,127 @@ public void localFailTest() {
     assertTrue("Operation should be done", ot.isDone());
   }
 
+  /**
+   * Test put operation with both dynamic success target and get replica by state enabled.
+   * Test cases:
+   * Case 1: 1 LEADER and 2 STANDBY replicas (regular case)
+   * Case 2: 1 LEADER and 3 STANDBY replicas. One of them returns REQUEST_DISABLED error code. There are two combinations
+   *         for remaining 3 replicas:
+   *         (1) two success and one failure (whole operation should succeed)
+   *         (2) one success and two failure (whole operation should fail)
+   * Case 3: 1 LEADER and 5 STANDBY replicas. Several combinations to discuss:
+   *         (1) 3 REQUEST_DISABLED, 2 success and 1 failure (operation should succeed)
+   *         (2) 2 REQUEST_DISABLED, 2 failure (operation should fail no matter what results are from remaining replicas)
+   *         (3) 1 REQUEST_DISABLED, 1 failure, 4 success (operation should succeed)
```

Comment:
should cases where we have 6 replicas but 0 replicas return REQUEST_DISABLED be tested in this test?

Reply:
added